### ISSUE TITLE
A non-numeric value encountered in

### ISSUE THREAD

**Issue by dataforxyz** [btcpayserver/woocommerce-plugin#13]
Best I can tell this is an issue with php 7.1. Is there a fix in the works? Should I make one?

PHP message: PHP Warning:  A non-numeric value encountered in /var/www/site/www/wp-content/plugins/btcpay-for-woocommerce/class-wc-gateway-bitpay.php on line 885" while reading response header from upstream, client: 141.101.77.153, server: site.net, request: "POST /wc-api/WC_Gateway_Bitpay/ HTTP/1.1", upstream: "fastcgi://unix:/run/phpfpm-site.sock:", host: "site.net"
2018/11/12 16:16:34 [error] 337#337: *83248 FastCGI sent in stderr: "PHP message: PHP Warning:  A non-numeric value encountered in /var/www/site/www/wp-content/plugins/btcpay-for-woocommerce/class-wc-gateway-bitpay.php on line 660

**Comment by bitcoinshirt** [comment]
Haven't noticed this but pr is welcome. Maybe @Kukks knows more.

**Comment by bobsburgers** [comment]
> 
> 
> Haven't noticed (running php 7.3 though) this but pr is welcome. Maybe @Kukks knows more.
> We're actually looking at completely re-ramping the plugin if you have any experience maybe you can help Kukks out.

Yes, I was looking at some of the other issues. If there are some specific tasks I would be happy to help. Ill keep watch for some I can do.

**Comment by bitcoinshirt** [comment]
For now, we really want to add our plugin to WordPress repository. There are some things that need to be cleaned. I'm sure @Kukks can tell you more.
For now check [this issue](https://github.com/btcpayserver/btcpayserver/issues/377) [and this one ](https://github.com/btcpayserver/woocommerce-plugin/issues/7)for specific things we have to fix in order to be accepted by the WP.

**Comment by Kukks** [comment]
Heya @bobsburgers  
I'm *almost* done with getting the plugin ready for wordpress plugin repo. Once that's done I would love to get rid of some of the obscure tooling in the project( no joke, we have to run grunt to move files around that requires ancient npm versions that needs to run in a docker that needs to be run through a powershell file lol) I am fairly confident we can clear most of the tooling without issue. :)

**Comment by NicolasDorier** [comment]
yeah took me quite a lot of time to be able to "compile" the project reliably. Needed to isolate the mess inside a dockerfile.
